classes/Cart.php

```diff
@@ -5113,4 +5061,107 @@ public function getTaxCountry(): Country
 
         return new Country($taxAddress->id_country);
     }
+
+    /**
+     * Alter raw cart details to adapt to display use case.
+     *
+     * @param array $summary
+     * @param bool $refresh
+     *
+     * @return array
+     */
+    private function alterSummaryForDisplay(array $summary, bool $refresh = false): array
+    {
+        $context = Context::getContext();
+        $currency = new Currency($this->id_currency);
+
+        $products = $this->getProducts($refresh);
```

Comment:
When calling `getSummaryDetails()`, `$this->getProducts()` will be called twice, and potentially, it will also be refreshed twice if `$refresh` is true.

Reply:
Maybe the best solution is to have two complete, independent implementations for getSummaryDetails() and getRawSummaryDetails(). It could be helpful to note exactly what's modified in `getSummaryDetails`.